integration/singularity/cleanup.go

```diff
@@ -0,0 +1,110 @@
+package singularity
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/filecoin-project/motion/blob"
+)
+
+type cleanupSchedulerConfig struct {
+	interval time.Duration
+}
+
+// This is run by the cleanup scheduler to determine whether to clean up a local
+// file.
+type cleanupReadyCallback func(ctx context.Context, blobID blob.ID) (bool, error)
+
+type cleanupScheduler struct {
+	cfg          cleanupSchedulerConfig
+	local        *blob.LocalStore
+	cleanupReady cleanupReadyCallback
+	closing      chan struct{}
+	closed       sync.WaitGroup
+}
+
+func newCleanupScheduler(
+	cfg cleanupSchedulerConfig,
+	local *blob.LocalStore,
+	cleanupReady cleanupReadyCallback,
+) *cleanupScheduler {
+	return &cleanupScheduler{
+		cfg:          cfg,
+		local:        local,
+		cleanupReady: cleanupReady,
+		closing:      make(chan struct{}),
+	}
+}
+
+func (cs *cleanupScheduler) start(ctx context.Context) {
+	cs.closed.Add(1)
+
+	go func() {
+		defer cs.closed.Done()
+
+		ctx, cancel := context.WithCancel(context.Background())
+		go func() {
+			<-cs.closing
+			cancel()
+		}()
+
+		ticker := time.NewTicker(cs.cfg.interval)
```

Comment:
Move this until after the first call to `cs.clenaup(ctx)`, to line 57, so that the timer does not start until after the first cleanup is done.

Reply:
I did it in this order because after the initial call, the behavior of the ticker is to tick `x duration` after the start of the previous cleanup, not after the end of the previous cleanup. so this way the behavior stays consistent with the regular loop, and cleanups will happen at predictable times.

if it is desired to start waiting for the duration after the previous cleanup finishes instead of immediately after it starts, we would probably have to replace the ticker with a timer that is manually scheduled at the end of a cleanup. lmk what u think

Reply:
OK, I see.

Would be nice to put the `defer ticker.Stop()` right after the ticker is created to make clear it is stopped at exit.